cloud/libvirt/actuators/machine/utils/volume.go

```diff
@@ -171,6 +158,17 @@ func CreateVolume(volumeName, poolName, baseVolumeID, source, volumeFormat strin
 		}
 
 		// create the volume
+		pool, err := virConn.LookupStoragePoolByName(poolName)
+		defer pool.Free()
```

Comment:
Can you actually call `pool.Free()` in the face of an error? Won't `pool` be nil in those cases?